README.md
EC-CUBEアフィリナビトラッキングプラグイン
=========================================
自社運営型アフィリエイトシステムのアフィリナビが提供する
売上トラッキングタグの設置と設定を自動化するプラグインです。


対応製品
--------
現在以下の製品・機能に対応しています。

* AN7
    * 売上トラッキング
* アフィリナビPRO
    * 売上トラッキング


インストール方法
----------------
1. [プロジェクトをダウンロード](https://github.com/zenith6/eccube-antracking/archive/master.tar.gz "eccube-antracking-master.tar.gz")します。

    ```bash
    # プロジェクトをダウンロードする
    wget https://github.com/zenith6/eccube-antracking/archive/master.tar.gz
    ```
2. ダウンロードした書庫をインストール可能なディレクトリレイアウトに修正します。
    eccube-antracking-master 以下のファイルが、書庫のルートディレクトリへ配置されるようにして下さい。

    ```bash
    # 書庫を展開する
    tar xzvf master.tar.gz
    # 展開先の eccube-antracking-master に移動する
    cd eccube-antracking-master
    # eccube-antracking-master 以下のファイルを AnTracking.tar.gz として再圧縮する
    tar czvf ../AnTracking.tar.gz *
    ```

3. EC-CUBEの管理画面にログインし、オーナーズストア > プラグインの管理 と進みます。
4. プラグイン登録にて、2. で作成した書庫を指定しインストールします。
5. アフィリナビトラッキングプラグインのプラグイン設定を開き、各種項目を入力します。
6. プラグインを有効にします。


プラグイン設定
--------------
ご利用の製品毎に設定項目が変わります。
次の手順に従って入力して下さい。

### AN7をご利用の場合
1. AN7にログインしキャンペーンのトラッキングタグ画面を開き、売上トラッキングタグを取得します。
   ユーザー定義パラメーター「注文番号」にEC-CUBEの受注管理で使用する注文番号が記録されます。
   ※「注文番号」を定義していない場合でも記録されます。
2. EC-CUBEの管理画面にログインし、プラグインの管理を開きます。
3. アフィリナビトラッキングプラグインのプラグイン設定を開きます。
4. 『ご利用の製品』を「アフィリナビPRO」にします。
5. 1. で取得した売上トラッキングタグを「売上トラッキングタグ」に入力します。
6. [登録する] をクリックして設定を保存します。
7. プラグインの管理に戻り、アフィリナビトラッキングプラグインの『有効』をチェックします。

### アフィリナビPROをご利用の場合
1. アフィリナビPROの管理画面にログインし、売上トラッキングタグを取得します。
2. EC-CUBEの管理画面にログインし、プラグインの管理を開きます。
3. アフィリナビトラッキングプラグインのプラグイン設定を開きます。
4. 『ご利用の製品』を「アフィリナビPRO」にします。
5. 1. で取得した売上トラッキングタグを「売上トラッキングタグ」に入力します。
6. [登録する] をクリックして設定を保存します。
7. プラグインの管理に戻り、アフィリナビトラッキングプラグインの『有効』をチェックします。


トラッキングについて
-------------------
ご利用の製品毎にトラッキングされる内容が変わります。

### AN7をご利用の場合

#### 売上トラッキング
トラッキングデータにバナー売上達成が記録されます。
トラッキングデータが解析された時点でバナー売上費用コミッションが発生します。
バナー売上費用コミッションのユーザー定義パラメーター「注文番号」に
EC-CUBEの受注管理で使用する注文番号が記録されます。

### アフィリナビPROをご利用の場合

#### 売上トラッキング
報酬管理の報酬確定処理に注文のトランザクションが記録されます。
トランザクション承認画面のデータ欄に
EC-CUBEの受注管理で使用する注文番号が記録されます。


ラインセス
---------
LGPLでご利用可能です。


お問い合わせ先
--------------
アフィリナビに関するお問い合わせはアフィリナビ公式サイト
[affilinavi.com](http://affilinavi.com/ "affilinavi.com") にて受け付けております。


ChangeLog.md
## 1.1.0 (2013-05-15)

AN7の仕様変更に対応しました。

* AN7のランディングトラッキングタグ追加に対応


## 1.0.0 (2013-04-03)

最初のリリース。


config.php
<?php
/*
 * アフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

require_once PLUGIN_UPLOAD_REALDIR .  'AnTracking/pages/LC_Page_Plugin_AnTracking_Config.php';

$page = new LC_Page_Plugin_AnTracking_Config();
register_shutdown_function(array($page, 'destroy'));
$page->init();
$page->process();


AnTracking.php
<?php
/*
 * EC-CUBEアフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/**
 * プラグインのメインクラス
 *
 * @package AnTracking
 * @author M-soft
 * @version $Id: $
 */
class AnTracking extends SC_Plugin_Base {
    const SESSION_NAME = 'antracking.trackingdata';

    /**
     * プラグイン設定
     * 
     * @var stdClass
     */
    static $settings;
    
    /**
     * コンストラクタ
     */
    public function __construct(array $arrSelfInfo) {
        parent::__construct($arrSelfInfo);
        
        // オートローダー用にライブラリへのパスを追加。
        ini_set('include_path', ini_get('include_path') . PATH_SEPARATOR . PLUGIN_UPLOAD_REALDIR . '/AnTracking/library');
    }
    
    /**
     * プラグインをインストールします。
     *
     * @param array $info プラグイン情報(dtb_plugin)
     * @return void
     */
    public function install($info) {
        // プラグインロゴを配置。
        copy(PLUGIN_UPLOAD_REALDIR . "AnTracking/logo.png", PLUGIN_HTML_REALDIR . "AnTracking/logo.png");
    }
    
    /**
     * プラグインをアンインストールします。
     *
     * @param array $info プラグイン情報
     * @return void
     */
    public function uninstall($info) {
        // プラグイン用のHTMLディレクトリを削除。
        SC_Helper_FileManager_Ex::deleteFile(PLUGIN_HTML_REALDIR . 'AnTracking');
    }
    
    /**
     * プラグインを有効化します。
     *
     * @param array $info プラグイン情報
     * @return void
     */
    public function enable($info) {
    }
    
    /**
     * プラグインを無効化します。
     *
     * @param array $info プラグイン情報
     * @return void
     */
    public function disable($info) {
    }
    
    /**
     * フックを登録します。
     *
     * @param SC_Helper_Plugin $plugin
     * @param int $priority
     */
    public function register(SC_Helper_Plugin $plugin, $priority) {
        parent::register($plugin, $priority);
        
        // デフォルトデバイス、スマートフォン向けのページにランディングトラッキングタグを挿入する。
        $plugin->addAction('prefilterTransform', array(&$this, 'hookActionInsertLandTrackingTag'));
        
        // 注文完了時に売上トラッキングタグ用のトラッキングデータを用意する。
        $plugin->addAction('LC_Page_Shopping_Confirm_action_confirm', array(&$this, 'hookActionPrepareSellTrackingData'));
        
        // 注文完了時の出力に売上トラッキングタグを挿入する。
        $plugin->addAction('outputfilterTransform', array(&$this, 'hookActionOutputSellTrackingTag'));
    }
    
    /**
     * フックアクション。デフォルトデバイス、スマートフォン向けのページにランディングトラッキングタグを挿入します。
     * 
     * @param string $source
     * @param LC_Page_Ex $page
     * @param string $filename
     */
    public function hookActionInsertLandTrackingTag(&$source, LC_Page_Ex $page, $filename) {
        // 挿入対象ページは site_main.tpl、デバイスはPC及びスマートフォン。
        $insert = in_array(basename(DIRECTORY_SEPARATOR . $filename), array('site_main.tpl'));
        $insert = $insert && in_array($page->arrPageLayout['device_type_id'], array(DEVICE_TYPE_PC, DEVICE_TYPE_SMARTPHONE));
        if (!$insert) {
            return;
        }
        
        $tracker = $this->getTracker();
        $data = array();
        $tag = $tracker->buildLandTrackingTag($data);
        
        // BODY最後尾を挿入する。
        $source =  str_replace('</body>', $tag . '</body>', $source);
    }
    
    /**
     * フックアクション。注文完了時に売上トラッキングタグ用のトラッキングデータを用意します。
     * 
     * @param LC_Page_EX $page
     */
    public function hookActionPrepareSellTrackingData(LC_Page_EX $page) {
        $order_id = $page->arrForm['order_id'];
        
        $context = (object)array(
            'page' => $page,
        );
        $tracker = $this->getTracker();
        $data = $tracker->buildSellTrackingData($context);
        $this->saveTrackingData('sell', $data);

        return $data;
    }
    
    /**
     * フックアクション。注文完了時に売上トラッキングタグを出力します。
     * 
     * @param string $source
     * @param LC_Page_EX $page
     * @param string $filename
     */
    public function hookActionOutputSellTrackingTag(&$source, LC_Page_EX $page, $filename) {
        // 注文完了ページ以外は無用。
        if (!($page instanceof LC_Page_Shopping_Complete)) {
            return;
        }
        
        $data = $this->loadTrackingData('sell');
        
        // トラッキングデータがなければ何もしない。
        if (!$data) {
            return;
        }

        $tracker = $this->getTracker();
        $tag = $tracker->buildSellTrackingTag($data);
        
        // 売上トラッキングタグを挿入する。
        $transfomer = new SC_Helper_Transform_Ex($source);
        $transfomer->select('body')->appendChild($tag);
        
        $source = $transfomer->getHTML();
        
        // 再表示時に売上トラッキングタグを再度出力しないようにするため。
        $this->deleteTrackingData('sell');
    }
    
    /**
     * @throws LogicException
     * @return AN_Tracker
     */
    public function getTracker() {
        $settings = AnTracking::loadSettings();
        switch ($settings->product) {
            case 'an7':
                $class = 'AN_An7Tracker';
                break;
                
            case 'anpro':
                $class = 'AN_AnproTracker';
                break;
                
            case 'anpro_st':
                $class = 'AN_AnproStTracker';
                break;
                
            default:
                throw new LogicException('Not supported product.'); 
        }

        $tracker = new $class($settings);
        return $tracker;
    }
    
    protected function saveTrackingData($type, $data) {
        $_SESSION[self::SESSION_NAME][$type] = $data;
    }
    
    protected function loadTrackingData($type) {
        return $_SESSION[self::SESSION_NAME][$type];
    }
    
    protected function deleteTrackingData($type = null) {
        if ($type === null) {
            unset($_SESSION[self::SESSION_NAME]);
        } else {
            unset($_SESSION[self::SESSION_NAME][$type]);
        }
    }
    
    /**
     * @return stdClass
     */
    public static function loadSettings($refresh = false) {
        if (!self::$settings || $refresh) {
            $query = SC_Query::getSingletonInstance();
            $row = $query->getCol('free_field1', 'dtb_plugin', 'plugin_code = ?', array(__CLASS__));
            if (empty($row[0])) {
                $settings = new stdClass();
            } else {
                $settings = json_decode($row[0]);
            }
            
            self::$settings = $settings;
        }
        
        return self::$settings;
    }

    /**
     * @param stdClass $settings
     */
    public static function saveSettings($settings) {
        $query = SC_Query::getSingletonInstance();
        $values = array();
        $values['free_field1'] = json_encode($settings);
        $query->update('dtb_plugin', $values, 'plugin_code = ?', array(__CLASS__));
        
        self::$settings = $settings;
    }
}


plugin_update.php
<?php
/*
 * EC-CUBEアフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/**
 * プラグインのプラグインアップデート
 *
 * @package AnTracking
 * @author M-soft
 * @version $Id: $
 */
class plugin_update {
    /**
     * プラグインをアップデートします。
     *
     * @param array $info プラグイン情報の連想配列(dtb_plugin)
     * @return void
     */
    function update($info) {
        $new_info = self::invokeUpTo($info);
        self::savePluginInfo($new_info, $info);
    }
    
    /**
     * プラグインの現在のバージョンに合わせて必要なアップデートを実行します。
     * 
     * @param array $info プラグイン情報の連想配列(dtb_plugin)
     * @return array アップデート後のプラグイン情報の連想配列(dtb_plugin)
     */
    protected static function invokeUpTo($info) {
        $points = self::getMigrationPoints();
        foreach ($points as $point) {
            if (version_compare($info['plugin_version'], $point['version'], '<')) {
                $info = call_user_func($point['up_to'], $info);
            }
        }
        
        return $info;
    }
    
    private static function getMigrationPoints() {
        $points = array();
        $sortkey = array();
        $class = new ReflectionClass(__CLASS__);
        $methods = $class->getMethods();
        $prefix_len = strlen('upTo_');
        foreach ($methods as $method) {
            if (strncmp($method->name, 'upTo_', $prefix_len) == 0) {
                $version = substr($method->name, $prefix_len);
                $version = strtr($version, array('_' => '.'));
                $points[$version] = array(
                    'version' => $version,
                    'up_to' => array($method->class, $method->name),
                );
                $sortkey[] = $version;
            }
        }
        
        natsort($sortkey);
        $sorted = array();
        foreach ($sortkey as $key) {
            $sorted[$key] = $points[$key];
        }
        return $sorted;
    }
    
    /**
     * プラグイン情報を保存します。
     * 
     * @param array $new_info 新しいプラグイン情報の連想配列(dtb_plugin)
     * @param array $old_info 以前のプラグイン情報の連想配列(dtb_plugin)
     */
    protected static function savePluginInfo($new_info, $old_info) {
        $query = SC_Query::getSingletonInstance();
        $fields = array(
            'plugin_name',
            'plugin_code',
            'class_name',
            'author',
            'author_site_url',
            'plugin_site_url',
            'plugin_version',
            'compliant_version',
            'plugin_description',
            'free_field1',
            'free_field2',
            'free_field3',
            'free_field4',
        );
        $values = array();
        foreach ($fields as $field) {
            $values[$field] = $new_info[$field];
        }
        $query->update('dtb_plugin', $values, 'plugin_code = ?', array($old_info['plugin_code']));
    }
    
    /**
     * バージョン 1.1.0 へ更新。
     * 
     * @param array $info プラグイン情報の連想配列(dtb_plugin)
     * @return array 更新後のプラグイン情報の連想配列(dtb_plugin)
     */
    protected static function upTo_1_1_0($info) {
        $info['plugin_version'] = '1.1.0';
        
        $settings = json_decode($info['free_field1']);
        $settings->an7_land_tracking_tag = '';
        $info['free_field1'] = json_encode($settings);
        
        return $info;
    }
}


library/AN/Tracker.php
<?php
/*
 * EC-CUBEアフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/**
 * トラッカーの基礎
 *
 * @package AnTracking
 * @author M-soft
 * @version $Id: $
 */
abstract class AN_Tracker {
    /**
     * @var stdClass プラグイン設定
     */
    protected $settings;
    
    public function __construct($settings = null) {
        if ($settings === null) {
            $settings = AnTracking::loadSettings();
        }
        
        $this->settings = $settings;
    }
    
    /**
     * @param array $data ランディングトラッキングデータ
     * @return string ランディングトラッキングタグ
     */
    abstract public function buildLandTrackingTag(array $data);

    /**
     * @param stdClass $context 実行環境
     */
    abstract public function buildLandTrackingData(stdClass $context);
    
    /**
     * @param array $data 売上トラッキングデータ
     * @return string 売上トラッキングタグ
     */
    abstract public function buildSellTrackingTag(array $data);
    
    /**
     * @param stdClass $context 実行環境
     */
    abstract public function buildSellTrackingData(stdClass $context);
}


library/AN/An7Tracker.php
<?php
/*
 * EC-CUBEアフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/**
 * AN7用トラッカー
 *
 * @package AnTracking
 * @author M-soft
 * @version $Id: $
 */
class AN_An7Tracker extends AN_Tracker {
    /**
     * @param array $data ランディングトラッキングデータ
     * @return string ランディングトラッキングタグ
     */
    public function buildLandTrackingTag(array $data) {
        return $this->settings->an7_land_tracking_tag;
    }

    /**
     * @param stdClass $context 実行環境
     */
    public function buildLandTrackingData(stdClass $context) {
        return array();
    }
    
    /**
     * @param stdClass $context 実行環境
     */
    public function buildSellTrackingData(stdClass $context) {
        $page = $context->page;
        $order_id = $page->arrForm['order_id'];

        $query = SC_Query::getSingletonInstance();
        $order = $query->getRow('*', 'dtb_order', 'order_id = ?', array($order_id));
        
        $data = array(
            'sales' => $order['subtotal'],
            'order_id' => $order_id,
        );
        return $data;
    }
    
    /**
     * @param array $data 売上トラッキングデータ
     * @return string 売上トラッキングタグ
     */
    public function buildSellTrackingTag(array $data) {
        $template = $this->settings->an7_sell_tracking_tag;
        $patterns = $replacements = array();
        
        // 売上金額
        $patterns['sales'] = '/(?<=\\s)params\\["sales"\\]\\s*=[^;]+;/mu';
        $replacements['sales'] = 'params["sales"] = ' . json_encode($data['sales']) . ';';
        
        // 注文番号（ユーザー定義パラメーター「注文番号」が存在する場合に置換）
        $patterns['order_id'] = '/(?<=\\s)params\\["custom\\[\\\\u6ce8\\\\u6587\\\\u756a\\\\u53f7\\]"\\]\\s*=[^;]+;/mu';
        $replacements['order_id'] = 'params["custom[\\u6ce8\\u6587\\u756a\\u53f7]"] = ' . json_encode($data['order_id']) . ';';

        // 注文番号（ユーザー定義パラメーター「注文番号」が存在しない場合、ユーザー定義パラメーターを追加する）
        if (!preg_match($patterns['order_id'], $template)) {
            $patterns[] = '/(?=var tracking_url =)/u';
            $replacements[] = $replacements['order_id'] . "\n\n";
        }
        
        $tag = preg_replace($patterns, $replacements, $template);
        return $tag;
    }
}


library/AN/AnproTracker.php
<?php
/*
 * EC-CUBEアフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/**
 * アフィリナビPRO用トラッカー
 *
 * @package AnTracking
 * @author M-soft
 * @version $Id: $
 */
class AN_AnproTracker extends AN_Tracker {
    /**
     * @param array $data ランディングトラッキングデータ
     * @return string ランディングトラッキングタグ
     */
    public function buildLandTrackingTag(array $data) {
        return '';
    }

    /**
     * @param stdClass $context 実行環境
     */
    public function buildLandTrackingData(stdClass $context) {
        return array();
    }
    
    /**
     * @param stdClass $context 実行環境
     */
    public function buildSellTrackingData(stdClass $context) {
        $page = $context->page;
        $order_id = $page->arrForm['order_id'];

        $query = SC_Query::getSingletonInstance();
        $order = $query->getRow('*', 'dtb_order', 'order_id = ?', array($order_id));
        
        $data = array(
            'sales' => $order['subtotal'],
            'data' => '注文番号#' . $order_id,
        );
        
        return $data;
    }
    
    /**
     * @param array $data 売上トラッキングデータ
     * @return string 売上トラッキングタグ
     */
    public function buildSellTrackingTag(array $data) {
        $template = $this->settings->anpro_sell_tracking_tag;

        preg_match('#src=["\'](https?://[^\'"]+)["\']#u', $template, $captures);
        $endpoint = $captures[1];
        $url = parse_url($endpoint);

        $form = array(
            't' => 'action',
            'o' => 'i',
            'guid' => 'ON',
            'sales' => $data['sales'],
            'data' => $data['data'],
        );
        
        $new_endpoint = $url['scheme'] . '://' . $url['host'] . $url['path'] . '?' . http_build_query($form);
        $replacement = 'src="' . htmlspecialchars($new_endpoint, ENT_QUOTES, 'UTF-8') . '"';
        $tag = preg_replace('#src=["\']https?://[^\'"]+["\']#u', $replacement, $template);
        
        return $tag;
    }
}


plugin_info.php
<?php
/*
 * EC-CUBEアフィリナビトラッキングプラグイン
 * Copyright (C) 2013 M-soft All Rights Reserved.
 * http://m-soft.jp/
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/**
 * プラグイン の情報クラス.
 *
 * @package AnTracking
 * @author M-soft
 * @version $Id: $
 */
class plugin_info {
    /** プラグインコード(必須)：プラグインを識別する為キーで、他のプラグインと重複しない一意な値である必要がありま. */
    static $PLUGIN_CODE = "AnTracking";
    
    /** プラグイン名(必須)：EC-CUBE上で表示されるプラグイン名. */
    static $PLUGIN_NAME = "EC-CUBEアフィリナビトラッキングプラグイン";
    
    /** プラグインバージョン(必須)：プラグインのバージョン. */
    static $PLUGIN_VERSION = "1.1.0";
    
    /** 対応バージョン(必須)：対応するEC-CUBEバージョン. */
    static $COMPLIANT_VERSION = "2.12.0";
    
    /** 作者(必須)：プラグイン作者. */
    static $AUTHOR = "M-soft";
    
    /** 説明(必須)：プラグインの説明. */
    static $DESCRIPTION = "自社運営型アフィリエイトシステムのアフィリナビとの連携が簡単に可能になります。";
    
    /** プラグイン作者URL：プラグイン毎に設定出来るURL（説明ページなど） */
    static $AUTHOR_SITE_URL = "http://affilinavi.com/";
    
    /** クラス名(必須)：プラグインのクラス（拡張子は含まない） */
    static $CLASS_NAME = "AnTracking";
    
    /** フックポイント：フックポイントとコールバック関数を定義します */
    static $HOOK_POINTS = array();
    
    /** プラグインURL：プラグイン毎に設定出来るURL（説明ページなど） */
    static $PLUGIN_SITE_URL = "http://affilinavi.com/";
    
    /** ライセンス */
    static $LICENSE = "LGPL";
}
